.github/scripts/validate-version.sh

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+
+validate_version() {
+  local version="$1"
+  if [[ ! "$version" =~ ^[0-9]+\.[0-9]+\.[0-9]+(-pre.*)?$ ]]; then
+    echo "::error::Invalid version format in VERSION file. Expected: X.Y.Z or X.Y.Z-pre*"
```

Comment:
A more useful error message would also add the version that was provided.

Reply:
I agree, I will add it in another PR since I don't want to wait for CI to run through again.